### PR TITLE
MM-15224 Initial config for RN UI Automation - Appium for Android app

### PR DESCRIPTION
**Pull request by saturninoabril**
#### Summary
This is the initial config for RN UI Automation - Appium for Android app.

How to run E2E test:
1. `make build-android` - build Android app and verify that the `Mattermost.apk` is available at root directory on successful build.
2. `npm run e2e:android` - npm script to run E2E for Android

Packages used:
- `appium@1.12.1`
- `wd@1.11.2`
- `tape@4.10.1`
- `tap-spec@5.0.0`

#### Ticket Link
Jira ticket: [MM-15224](https://mattermost.atlassian.net/browse/MM-15224)

#### Checklist
- [x] Added E2E tests

#### Device Information
This PR was tested on: [Android]

### CODE REVIEW

File: e2e/script/e2e_init.js [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+let SERVER_PID;
+let APPIUM_PID;
+let exitCode;
+
+// packages
+const APPIUM = 'appium@1.12.1';
+const WD = 'wd@1.11.2';
+const TAPE = 'tape@4.10.1';
+const TAP_SPEC = 'tap-spec@5.0.0';
```
Comment (saturninoabril) [inline e2e/script/e2e_init.js]: Packages are installed only when running E2E test.

File: package.json [hunk 2]
```diff
@@ -100,8 +100,10 @@
     "react-dom": "16.8.2",
     "redux-mock-store": "1.5.3",
     "remote-redux-devtools": "0.5.16",
+    "shelljs": "0.8.3",
```
Comment (saturninoabril) [inline package.json]: Introduce `shelljs` package to easily run shell command in javascript and easily do control flow.

**Comment by saturninoabril** [comment]
@enahum @thekiiingbob This should now be ready for review.

@thekiiingbob Let's get this in before your changes to `webdriverio`.

File: e2e/script/e2e_init.js [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+} finally {
+    stopPackagerServer();
+    stopAppiumServer();
+}
+
+exit(exitCode);
+
+// Helper functions
+
+function describe(message) {
```
Comment (thekiiingbob) [inline e2e/script/e2e_init.js]: I'd rename this. I was confused when I saw it used above because I thought it was coming from mocha/jest.

Comment (saturninoabril) [reply e2e/script/e2e_init.js]: Changed to `logStep`.

**Comment by enahum** [comment]
@saturninoabril I ran this locally and I'm getting the following output

```
18:10 $ npm run e2e:android

> mattermost-mobile@1.19.0 e2e:android /Users/enahum/Projects/go/src/github.com/mattermost/mattermost-mobile
> node ./e2e/script/e2e_init.js --android --install


    ---------------------------------------
    Installing end-to-end framework
    ---------------------------------------
    
Started packager server, pid @24990
Started appium server, pid @25016

    ---------------------------------------
    Running E2E tests
    ---------------------------------------
    

  Setup and open Android app


    ✖ test exited without ending
    -----------------------------
      operator: fail
      at: process.<anonymous> (/Users/enahum/Projects/go/src/github.com/mattermost/mattermost-mobile/node_modules/tape/index.js:90:19)
      stack: |-
  Error: test exited without ending
  at Test.assert [as _assert] (/Users/enahum/Projects/go/src/github.com/mattermost/mattermost-mobile/node_modules/tape/lib/test.js:226:54)
  at Test.bound [as _assert] (/Users/enahum/Projects/go/src/github.com/mattermost/mattermost-mobile/node_modules/tape/lib/test.js:77:32)
  at Test.fail (/Users/enahum/Projects/go/src/github.com/mattermost/mattermost-mobile/node_modules/tape/lib/test.js:319:10)
  at Test.bound [as fail] (/Users/enahum/Projects/go/src/github.com/mattermost/mattermost-mobile/node_modules/tape/lib/test.js:77:32)
  at Test._exit (/Users/enahum/Projects/go/src/github.com/mattermost/mattermost-mobile/node_modules/tape/lib/test.js:191:14)
  at Test.bound [as _exit] (/Users/enahum/Projects/go/src/github.com/mattermost/mattermost-mobile/node_modules/tape/lib/test.js:77:32)
  at process.<anonymous> (/Users/enahum/Projects/go/src/github.com/mattermost/mattermost-mobile/node_modules/tape/index.js:90:19)
  at process.emit (events.js:182:13)


    ✖ test exited without ending
    -----------------------------
      operator: fail
      at: process.<anonymous> (/Users/enahum/Projects/go/src/github.com/mattermost/mattermost-mobile/node_modules/tape/index.js:90:19)
      stack: |-
  Error: test exited without ending
  at Test.assert [as _assert] (/Users/enahum/Projects/go/src/github.com/mattermost/mattermost-mobile/node_modules/tape/lib/test.js:226:54)
  at Test.bound [as _assert] (/Users/enahum/Projects/go/src/github.com/mattermost/mattermost-mobile/node_modules/tape/lib/test.js:77:32)
  at Test.fail (/Users/enahum/Projects/go/src/github.com/mattermost/mattermost-mobile/node_modules/tape/lib/test.js:319:10)
  at Test.bound [as fail] (/Users/enahum/Projects/go/src/github.com/mattermost/mattermost-mobile/node_modules/tape/lib/test.js:77:32)
  at Test._exit (/Users/enahum/Projects/go/src/github.com/mattermost/mattermost-mobile/node_modules/tape/lib/test.js:191:14)
  at Test.bound [as _exit] (/Users/enahum/Projects/go/src/github.com/mattermost/mattermost-mobile/node_modules/tape/lib/test.js:77:32)
  at process.<anonymous> (/Users/enahum/Projects/go/src/github.com/mattermost/mattermost-mobile/node_modules/tape/index.js:90:19)
  at process.emit (events.js:182:13)


    ✖ test exited without ending
    -----------------------------
      operator: fail
      at: process.<anonymous> (/Users/enahum/Projects/go/src/github.com/mattermost/mattermost-mobile/node_modules/tape/index.js:90:19)
      stack: |-
  Error: test exited without ending
  at Test.assert [as _assert] (/Users/enahum/Projects/go/src/github.com/mattermost/mattermost-mobile/node_modules/tape/lib/test.js:226:54)
  at Test.bound [as _assert] (/Users/enahum/Projects/go/src/github.com/mattermost/mattermost-mobile/node_modules/tape/lib/test.js:77:32)
  at Test.fail (/Users/enahum/Projects/go/src/github.com/mattermost/mattermost-mobile/node_modules/tape/lib/test.js:319:10)
  at Test.bound [as fail] (/Users/enahum/Projects/go/src/github.com/mattermost/mattermost-mobile/node_modules/tape/lib/test.js:77:32)
  at Test._exit (/Users/enahum/Projects/go/src/github.com/mattermost/mattermost-mobile/node_modules/tape/lib/test.js:191:14)
  at Test.bound [as _exit] (/Users/enahum/Projects/go/src/github.com/mattermost/mattermost-mobile/node_modules/tape/lib/test.js:77:32)
  at process.<anonymous> (/Users/enahum/Projects/go/src/github.com/mattermost/mattermost-mobile/node_modules/tape/index.js:90:19)
  at process.emit (events.js:182:13)




  Failed Tests: There were 3 failures

    Setup and open Android app

      ✖ test exited without ending
      ✖ test exited without ending
      ✖ test exited without ending


  total:     3
  passing:   0
  failing:   3
  duration:  22.7s


Error thrown running E2E tests

    ---------------------------------------
    Failed: error encountered
    ---------------------------------------
    
Stopped appium 25016
Stopped packager 24990
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! mattermost-mobile@1.19.0 e2e:android: `node ./e2e/script/e2e_init.js --android --install`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the mattermost-mobile@1.19.0 e2e:android script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/enahum/.npm/_logs/2019-05-22T22_12_25_427Z-debug.log
✘-1 ~/Projects/go/src/github.com/mattermost/mattermost-mobile [e2e L|✔] 
18:12 $ 
```

**Comment by saturninoabril** [comment]
Thanks @enahum and sorry I missed your comment.

It's probably due to the Appium server not running on your local machine.  I forgot to mention in the description that it should be started manually (as of this PR) as I have not added yet the Appium service to do it automatically.

Do you have Appium installed?  If not yet, you may download it at https://github.com/appium/appium-desktop/releases/tag/v1.13.0.  Run the desktop app and then start the server.  Once running, you may initiate `npm run e2e:android`.

I'd be happy to connect to know more of the issue you're getting.

**Comment by saturninoabril** [comment]
Ticket to address it separately - https://mattermost.atlassian.net/browse/MM-15804

**Comment by enahum** [comment]
I could try running it manually but that defeat the purpose, in any case I think that should be part of the “initial” implementation

**Comment by enahum** [comment]
Talked with Saturn and once this is submitted against master it will include starting up and tearing down the appium server + the simulators

**Comment by saturninoabril** [comment]
Thanks @enahum, we'll ensure to address that 🎉

**Pull request merged**